internal-packages/no-nodejs-compat-flag-static-error-page/assets/index.html

```diff
@@ -0,0 +1,246 @@
+				gap: 2rem;
+			}
+
+			main > div {
+				flex: 1;
+			}
+
+			@media (width < 30rem) {
+				main {
+					flex-direction: column;
+				}
+			}
+
+			@media (width < 20rem) {
+				html {
+					font-size: 45%;
+				}
+			}
+
+			dialog {
+				width: 90vw;
+				height: 90vh;
+				display: flex;
+				flex-direction: column;
+				gap: 1rem;
+				border: 1px solid black;
+				border-radius: 10px;
+			}
+
+			dialog:not([open]) {
+				display: none;
+			}
+
+			dialog > div {
+				display: flex;
+				flex-direction: column;
+				overflow: auto;
+				padding: 1rem 2rem;
+			}
+
+			dialog > div * {
+				max-width: 100%;
+			}
+
+			dialog img {
+				margin: 1rem;
+				width: 90%;
+				display: block;
+			}
+
+			dialog > form {
+				align-self: flex-end;
+			}
+
+			dialog > form button {
+				font-size: 2.7rem;
+			}
+
+			dialog > form button:hover {
+				scale: 1.2;
+			}
+
+			button {
+				border: none;
+				background: none;
+				cursor: pointer;
+				padding: 0;
+			}
+
+			#more-details-btn {
+				font-style: italic;
+			}
+
+			#more-details-btn:hover {
+				text-decoration: underline;
+			}
+		</style>
+	</head>
+	<body>
+		<div>
+			<header>
+				<h1>
+					<span class="error">NodeJS Compatibility Error</span>
+					<span class="error-description"
+						>no <code>nodejs_compat</code> flag set</span
+					>
+				</h1>
+			</header>
+
+			<main>
+				<div class="what-happened">
+					<h2>What happened?</h2>
+					<p>
+						The page you've requested has been built using
+						<a
+							href="https://github.com/cloudflare/next-on-pages"
+							target="_blank"
+							>@cloudflare/next-on-pages</a
+						>, but hasn't been properly configured.
+					</p>
+				</div>
+
+				<div class="what-can-i-do">
+					<h2>What can I do?</h2>
+					<p>
+						<strong>If you are the owner of this website:</strong>
+						<br />
+						you should go to the Pages Project's
+						<a
+							href="https://dash.cloudflare.com/?to=/:account/pages/view/:pages-project/settings/functions#compatibility_flags_section"
+							target="_blank"
+							>compatibility flags settings section</a
+						>
+						and add the <code>compat_nodejs</code> flag to both your production
+						and preview deployments.
+					</p>
+					<p>
+						<button id="more-details-btn">More details &gt;</button>
+					</p>
+				</div>
+			</main>
+		</div>
+		<dialog>
+			<form method="dialog">
+				<button>&times;</button>
+			</form>
+			<div>
+				<p>
+					Access your Pages project and go to the
+					<a
+						href="https://dash.cloudflare.com/?to=/:account/pages/view/:pages-project/settings/functions"
+						target="_blank"
+						>function settings page</a
+					>.
+				</p>
+
+				<p>
+					Then in the compatibility flags section for both the production and
+					preview deployments type <code> nodejs_compat </code> in the input box
+					and click on save to apply the changes:
+					<img
+						src="https://no-nodejs-compat-flag-static-error-page.pages.dev/img/input.png"
```

Comment:
Note that you had to use the full URL here including host because this file is being rendered inside the broken next-on-pages deployment. But this means that if you decide to deploy it somewhere else, you need to remember to update these.
I wonder if you could reduce the duplication by setting a `base[href]` in the page?

Reply:
What about hosting the image in this GitHub repo and using the raw link to it @dario-piotrowicz?

Reply:
oh... yeah I didn't think of that! 😓

sounds like a good idea 🙂, thanks! 😄

Reply:
applied, thanks @james-elicx 😄 

(PS: unfortunately I cannot test this change until I actually merge this PR 😓)

Reply:
Sorry @james-elicx after chats with Pete and the comment from Greg I'm going to actually include the html in the user's app, it should practically have no impact on the application 🙂